src/katello/deb_tracer.py

```diff
@@ -27,45 +27,42 @@ def use_flag():
 
 
 def needrestart():
-    process = subprocess.Popen(["needrestart", "-b"], stdout=subprocess.PIPE)
-    output = process.communicate()
-    return output[0].decode().split("\n")
+    output = subprocess.check_output(["needrestart", "-b"])
+    return output.decode().split("\n")
 
 
 def use_needrestart():
     apps = []
     services = []
-    kasta = None
     output = needrestart()
     for item in output:
         output = item.split(":")
         if output[0] == "NEEDRESTART-KSTA":
-           ksta = output[1] 
+            ksta = output[1] 
         elif output[0] == "NEEDRESTART-SVC":
             services.append(output[1])
 
-    if int(ksta) == 2 or int(ksta) == 3:
+    services = list(map(lambda item: item.replace(".service", ""), services))
```

Comment:
I'd use a list comprehension if you're going to do this:
```suggestion
    services = [item.replace(".service", "") for item in services]
```

However, I'd change line 43 instead.

Reply:
I did it in the way @sbernhard suggested it